packages/fetchye/src/useFetchye.js

```diff
@@ -78,16 +75,12 @@ const useFetchye = (
       numOfRenders.current
     ),
     run() {
-      const runOptions = handleDynamicHeaders(options);
-      const runComputedKey = typeof options.headers === 'function'
-        ? computeKey(key, defaultMapOptionsToKey(mapOptionsToKey(runOptions)))
-        : computedKey;
       return runAsync({
         dispatch,
-        computedKey: runComputedKey,
+        computedKey,
```

Comment:
```suggestion
        computedKey: computeKey(key, options),
```

I think you still need to re-compute the key inside the run function.

This allows the `handleDynamicHeaders` function to re-call the users dynamic header function, re-creating their dynamic data as close to the `run` as possible